pkg/istio/processor/controller_test.go

```diff
@@ -0,0 +1,219 @@
+package processor
+
+import (
+	"fmt"
+	"github.com/stretchr/testify/assert"
+	"github.com/yahoo/k8s-athenz-istio-auth/pkg/istio/rbac/common"
+	"istio.io/api/rbac/v1alpha1"
+	"istio.io/istio/pilot/pkg/config/memory"
+	"istio.io/istio/pilot/pkg/model"
+	"k8s.io/api/core/v1"
+	"testing"
+)
+
+func newSrSpec() *v1alpha1.ServiceRole {
+	return &v1alpha1.ServiceRole{
+		Rules: []*v1alpha1.AccessRule{
+			{
+				Services: []string{common.WildCardAll},
+				Methods:  []string{"GET"},
+				Constraints: []*v1alpha1.AccessRule_Constraint{
+					{
+						Key:    common.ConstraintSvcKey,
+						Values: []string{"test-svc"},
+					},
+				},
+			},
+		},
+	}
+}
+
+func newSrbSpec(role string) *v1alpha1.ServiceRoleBinding {
+	return &v1alpha1.ServiceRoleBinding{
+		RoleRef: &v1alpha1.RoleRef{
+			Kind: common.ServiceRoleKind,
+			Name: role,
+		},
+		Subjects: []*v1alpha1.Subject{
+			{
+				User: "test-user",
+			},
+		},
+	}
+}
+
+func newCache() (model.ConfigStoreCache, error) {
+	configDescriptor := model.ConfigDescriptor{
+		model.ClusterRbacConfig,
+		model.ServiceRole,
+		model.ServiceRoleBinding,
+	}
+
+	c := memory.NewController(memory.Make(configDescriptor))
+	_, err := c.Create(func() model.Config {
+		return common.NewConfig(model.ServiceRole.Type, "test-ns", "test-role", newSrSpec())
```

Comment:
Do you think this would be more readable if we changed these one liner function calls to be a temporary variable which is the return from common.NewConfig which is then passed into the Create function? Maybe there's also a way to create two common objects at the top which can be reused in multiple places for service role and service role binding?